image-loader/app/controllers/ImageLoaderController.scala

```diff
@@ -148,38 +155,58 @@ class ImageLoaderController(auth: Authentication,
       )
 
       logger.info("importImage request start")
+      val existingImageStatus = uploadStatusTable.getStatus(uri.split("/").last).map {
+        case Some(Right(record)) => (record.uploadedBy, record.identifiers, record.fileName)
+        case _ => (None, None, None)
+      }.recover{case _ => (None, None, None)}
```

Comment:
If you make the whole record an option you can write these three lines as `_.flatMap(_.toOption)`